Enemies/Dodongo.cs

```diff
@@ -174,14 +149,18 @@ public void OnCollision(List<CollisionInfo> collisions)
                     IsColliding = false;
                     ChangeDirection();
                 }
-                else if (collidedWith == CollisionLayer.PlayerWeapon)
+                else if (collision.CollidedWith.Collidable is Explosion)
```

Comment:
This will be a good implementation aside from the logic in the enemy utilities that also handles the dodongo. I like this update here!